disasm-test/Main.hs

```diff
@@ -558,6 +562,45 @@ runCompileTest llvmVersion knownBugs sweet expct = do
                   -- .ll files are not compared; see runAssemblyTest for details.
 
 
+----------------------------------------------------------------------
+-- Pre-existing bitcode tests tests
+
+bitcodeCube :: VersionCheck -> TS.CUBE
+bitcodeCube llvmver = (assemblyCube llvmver)
+                        { TS.rootName = "*.bc"
+                        , TS.inputDirs = ["disasm-test/bc_src_tests"]
+                        , TS.sweetAdjuster = rangeMatch llvmver
+                        }
+
+runRawBCTest :: VersionCheck -> KnownBugs -> TS.Sweets -> TS.Expectation
+               -> IO [TestTree]
+runRawBCTest llvmVersion knownBugs sweet expct = do
+  shouldSkip <- skipTest expct
+  let tmod = if shouldSkip
+             then ignoreTestBecause "not valid for this LLVM version"
+             else case isKnownBug knownBugs sweet expct llvmVersion of
+                    Just (from, why) ->
+                      expectFailBecause $ why <> " [see " <> from <> "]"
+                    Nothing -> id
+  let pfx = TS.rootBaseName sweet
+  let bc = TS.rootFile sweet
+  return $ (:[]) $ tmod
+    $ testCaseM llvmVersion pfx
+    $ with2Files (parseBC pfx bc)
+        $ \(parsed1, ast) ->
+            case ast of
+              Nothing ->
+                -- No round trip, so this just verifies that the bitcode could be
+                -- parsed without generating an error.
+                return ()
+              Just ast1 ->
+                -- Assemble and re-parse the bitcode to make sure it can be
+                -- round-tripped successfully.
+                with2Files (processLL pfx parsed1)
+                $ \(_, Just ast2) -> diffCmp ast1 ast2
```

Comment:
This uni-pattern match on `Just` emits a warning with GHC 9.2 or later:

```
disasm-test/Main.hs:600:19: warning: [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a lambda abstraction:
        Patterns of type ‘(FilePath, Maybe FilePath)’ not matched:
            ([], Nothing)
            ((_:_), Nothing)
    |
600 |                 $ \(_, Just ast2) -> diffCmp ast1 ast2
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Actually, two other uses of `with2Files` in this file produce the same warning as well. Is it possible to restructure things to avoid this?

Reply:
Sure: fixed.